src/services/earn/earn-service.ts

```diff
@@ -137,6 +144,109 @@ export class EarnService implements IEarnService {
+        address: vault,
+        functionName: 'ownerOf',
+        args: [bigIntPositionId],
+      }),
+      this.checkIfNeedsSwap({ chainId, strategyId: bigIntPositionId, depositToken: increaseInfo.token }),
+    ]);
+    const callVaultDirectly = !increase || increaseInfo.amount === 0n || ('token' in increase && !needsSwap);
+
+    if (callVaultDirectly) {
+      // If don't need to use Permit2, then just call the vault
+      return {
+        to: vault,
+        data: encodeFunctionData({
+          abi: vaultAbi,
+          functionName: 'increasePosition',
+          args: [BigInt(positionId), increaseInfo.token as ViemAddress, increaseInfo.amount],
+        }),
+        value: increaseInfo.value,
+      };
+    }
+
+    // If we get to this point, then we'll use the Companion for the increase
+    const calls: Hex[] = [];
+
+    const recipient = needsSwap ? COMPANION_SWAPPER_CONTRACT.address(chainId) : companion;
+    if ('permitData' in increase!) {
+      // Handle take from caller (if necessary)
+      calls.push(buildTakeFromCallerWithPermit(increase.permitData, increase.signature, recipient));
+    } else if (!isSameAddress(increaseInfo.token, Addresses.NATIVE_TOKEN)) {
+      calls.push(buildTakeFromCaller(increaseInfo.token, increaseInfo.amount, recipient));
+    }
+
+    const depositToken = (needsSwap ? asset : increaseInfo.token) as ViemAddress;
+    let maxApprove = false;
+    const promises = [];
+
+    if (needsSwap) {
+      const swapPromise = await this.getSwapData({
+        request: {
+          chainId,
+          sellToken: increaseInfo.token,
+          buyToken: asset,
+          order: { type: 'sell', sellAmount: increaseInfo.amount },
+        },
+        leftoverRecipient: positionOwner,
+        swapConfig: increase?.swapConfig,
+      }).then(({ swapData }) => calls.push(swapData));
+      promises.push(swapPromise);
+    }
+
+    if (!isSameAddress(depositToken, Addresses.NATIVE_TOKEN)) {
+      const allowancePromise = this.allowanceService
+        .getAllowanceInChain({
+          chainId,
+          owner: companion,
+          spender: vault,
+          token: depositToken,
+        })
+        .then((allowance) => {
+          // Only if the allowance is 0 we need to max approve the vault
+          maxApprove = allowance === 0n;
+        });
+      promises.push(allowancePromise);
+    }
+
+    await Promise.all(promises);
+
+    // Handle permission permit
+    if (permissionPermit && 'signature' in increase) {
+      calls.push(buildPermissionPermit(bigIntPositionId, increase.signature as Hex, permissionPermit, vault));
+    }
+
+    // Handle increase
+    calls.push(
+      encodeFunctionData({
+        abi: companionAbi,
+        functionName: 'increasePosition',
+        args: [vault as ViemAddress, bigIntPositionId, asset, Uint.MAX_256, maxApprove],
+      })
+    );
+
+    // Build multicall and return tx
+    return buildCompanionMulticall({ chainId, calls, value: increaseInfo?.value });
```

Comment:
When would `increaseInfo` be undefined?
```suggestion
    return buildCompanionMulticall({ chainId, calls, value: increaseInfo.value });
```